src/doc/src/reference/build-scripts.md

```diff
@@ -250,6 +255,51 @@ identifier, the value should be a string.
 [conditional compilation]: ../../reference/conditional-compilation.md
 [option-cfg]: ../../rustc/command-line-arguments.md#option-cfg
 
+### `cargo::rustc-check-cfg=CHECK_CFG` {#rustc-check-cfg}
+
+This instruction is related to checking the conditional configurations of the
+crate at compile time. That is, checking that every _reachable_ `#[cfg]` matches
+the expected config names and values.
+
+The `rustc-check-cfg` instruction adds to the list of expected config names
+and values in addition to Cargo features and [well known cfgs][well-known-cfgs].
+
+It is backed by the [`--check-cfg` flag][option-check-cfg] of the compiler.
+
+The instruction can be used like this:
+
+```rust,no_run
+// build.rs
+println!("cargo::rustc-check-cfg=cfg(foo, values(\"bar\"))");
+```
+
+Note that all possible cfgs should be defined, regardless of which cfgs are
+currently enabled. This includes all possible values of a given cfg name.
+
+It is recommended to group the `cargo::rustc-check-cfg` and
+[`cargo::rustc-cfg`][option-cfg] instructions as closely as possible in order to
+avoid typos, missing check-cfg, stalled cfgs...
+
+<details>
```

Comment:
This will be the first place we use `<details>` element. I don't object to the usage here. Just feel like this example is quite succinct and doesn't need to wrap in `<details>`.

The alternative idea is that we might want to expand this [Conditional compilation](http://127.0.0.1:3000/reference/build-script-examples.html#conditional-compilation) example instead.

Reply:
No opinion on it either, but I can't see the example (localhost url).

Reply:
My bad 😅.
Here it is <https://doc.rust-lang.org/nightly/cargo/reference/build-script-examples.html#conditional-compilation>.

Reply:
Good point. I propose we do both, remove the `<details>` and update the build script example.

Reply:
Updated both.